stl/inc/ranges

```diff
@@ -4748,6 +4748,580 @@ namespace ranges {
     } // namespace views
 
 #if _HAS_CXX23
+    template <class _Size>
+    constexpr _Size _Div_ceil(_Size _Num, _Size _Denom) noexcept(
+        noexcept(_Num / _Denom) && noexcept(_Num % _Denom) && noexcept(++_Num)) {
+        _Size _Result = _Num / _Denom;
+        if (_Num % _Denom) {
+            ++_Result;
+        }
+        return _Result;
+    }
+
+    template <input_range _Vw>
+        requires view<_Vw>
+    class chunk_view : public view_interface<chunk_view<_Vw>> {
+    private:
+        /* [[no_unique_address]] */ _Vw _Range{};
+        range_difference_t<_Vw> _Count     = 0;
+        range_difference_t<_Vw> _Remainder = 0;
+        _Non_propagating_cache<iterator_t<_Vw>> _Current{};
+
+        class _Outer_iterator {
+        private:
+            friend chunk_view;
+            chunk_view* _Parent{};
+
+            constexpr explicit _Outer_iterator(chunk_view& _Parent_) noexcept : _Parent(_STD addressof(_Parent_)) {}
+
+            _NODISCARD constexpr bool _Is_end() const
+                noexcept(noexcept(_Implicitly_convert_to<bool>(*_Parent->_Current == _RANGES end(_Parent->_Range)))) {
+                return *_Parent->_Current == _RANGES end(_Parent->_Range) && _Parent->_Remainder != 0;
+            }
+
+            _NODISCARD constexpr range_difference_t<_Vw> _Get_size() const
+                noexcept(noexcept(_Div_ceil(_RANGES end(_Parent->_Range) - *_Parent->_Current, _Parent->_Count))) {
+                const auto _Size = _RANGES end(_Parent->_Range) - *_Parent->_Current;
+                if (_Size < _Parent->_Remainder) {
+                    return _Size == 0 ? 0 : 1;
+                }
+                return _Div_ceil(_Size - _Parent->_Remainder, _Parent->_Count) + 1;
+            }
+
+
+        public:
+            using iterator_concept = input_iterator_tag;
+            using difference_type  = range_difference_t<_Vw>;
+
+            struct value_type : public view_interface<value_type> {
+            private:
+                friend _Outer_iterator;
+                chunk_view* _Parent{};
+
+                class _Inner_iterator {

```

Comment:
This should be a member of `chunk_view`, in my opinion. `value_type` _could_ be placed into `chunk_view`, but then the type name wouldn't make sense (it is the `value_type` of `_Outer_iterator`)

Reply:
I mean it is an exposition only member but I pulled it out to be more consistent with the standard